Proofread on *.md

ickc opened the pull request:
Proofreading *.md files (i.e. excluding `MANUAL.txt` for now). The commit messages should be self-explanatory. They mainly are:

- add relevant links
- proofread on grammar, syntax, etc.
- consistency

Feel free to squash this into 1 commit. I want them to be more atomic in case you want to cherry-pick.

jgm commented on CONTRIBUTING.md:
```diff
@@ -47,13 +47,13 @@ a bug in pandoc.  Quoting from the MANUAL:
+For example, both `docx` and `odt` formats can represent margin size, but
+because pandoc's internal document model does not contain a representation of
 margin size, this information will be lost on converting from docx
-to odt.  (You can, however, customize margin size using `--reference-odt`.)
```
Actually this should now be `--reference-doc`!  Didn't notice tihs before.

jgm commented on CONTRIBUTING.md:
```diff
@@ -47,13 +47,13 @@ a bug in pandoc.  Quoting from the MANUAL:
+to `odt`.  (You can, however, customize margin size using `--reference-odt`.)
 
 So before submitting a bug report, consider whether it might be
-"out of scope."  If it concerns a feature of documents that isn't
```
This is actually the correct punctuation in American style (period inside quote).
It's illogical, I know.

jgm commented on doc/using-the-pandoc-api.md:
```diff
@@ -313,10 +313,10 @@ Note that this looks first in the
 users to override the system defaults.  If you want to disable
 this behavior, use `setUserDataDir Nothing`.
 
-To render a template, use `renderTemplate'`, which takes two
```
ACtually it is `renderTemplate'`, the prime is there in the function name.
(There's also `renderTemplate`, which is different.)

jgm commented on doc/using-the-pandoc-api.md:
```diff
@@ -313,10 +313,10 @@ Note that this looks first in the
+To render a template, use `renderTemplate`, which takes two
 arguments, a template (String) and a context (any instance
 of ToJSON).  If you want to create a context from the metadata
-part of a Pandoc document, use `metaToJSON'` from
```
It really is `metaToJSON'`.

jgm commented:
Thanks!  I've made a few comments where changes are needed.

ickc commented:
I added 3 commits to fix the things you commented (never knew the rules about that punctuation in American style, perhaps it is different in British English, that I was grown up with).

I added 3 more commits to different fixes, where the 0b09409 is about updating the years in the copyright. I checked the commit history of them to see they have been edited in the last 2 years. There are some cases like `src/Text/Pandoc/Readers/Haddock.hs` that I don't know how to update the copyright, because originally it was copyrighted by an author and later updated by others.
